From 0e382a4ec10cde85304d5dff0ff213118f111ec0 Mon Sep 17 00:00:00 2001
From: Catalin Enache <catalin.enache@windriver.com>
Date: Sat, 26 Dec 2020 10:32:24 +0200
Subject: [PATCH] Revert "Omit to optimize vsprintf.c/kasprintf.c"

commit 7a70b2da48e89f4dab0a443cac0a2a3f7ec6d49d from linux-yocto

This reverts commit 0817a77d8c65f150d659f0a38b5f108372f381e6.

We no longer need to override optimization levels for arm on
vsprintf and asprintf. There are no more boot/build issues, and
in fact, the different optimization level is triggering build
warnings.

This reverts commit 0817a77d8c65f150d659f0a38b5f108372f381e6.

Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
[CE: backport of 7a70b2da48e8 from v5.4/standard/arm-versatile-926ejs branch of linux-yocto]
Signed-off-by: Catalin Enache <catalin.enache@windriver.com>
---
 lib/Makefile | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/lib/Makefile b/lib/Makefile
index 0cdf5732cf33..87b3be3c336d 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -99,15 +99,11 @@ CFLAGS_kobject.o += -DDEBUG
 CFLAGS_kobject_uevent.o += -DDEBUG
 endif
 
-
 obj-$(CONFIG_DEBUG_INFO_REDUCED) += debug_info.o
 CFLAGS_debug_info.o += $(call cc-option, -femit-struct-debug-detailed=any)
 
 obj-y += math/ crypto/
 
-CFLAGS_vsprintf.o += -O0
-CFLAGS_asprintf.o += -O0
-
 obj-$(CONFIG_GENERIC_IOMAP) += iomap.o
 obj-$(CONFIG_GENERIC_PCI_IOMAP) += pci_iomap.o
 obj-$(CONFIG_HAS_IOMEM) += iomap_copy.o devres.o
-- 
2.26.1

